pennylane/ops/op_math/condition.py

```diff
@@ -430,3 +446,128 @@ def wrapper(*args, **kwargs):
+            if pred:
+                return run_jaxpr(jaxpr_elif, *args)
+            return elif_branch(args, rest_preds, rest_jaxpr_elifs)
+
+        def false_branch(args):
+            if jaxpr_false is not None:
+                return run_jaxpr(jaxpr_false, *args)
+            return ()
+
+        if condition:
+            return true_branch(args)
+        if elifs_conditions.size > 0:
+            return elif_branch(args, elifs_conditions, jaxpr_elifs)
+        return false_branch(args)
+
+    @cond_prim.def_abstract_eval
+    def _(*_, jaxpr_true, jaxpr_false, jaxpr_elifs):
+
+        # We check that the return values in each branch (true, and possibly false and elifs)
+        # have the same abstract values.
+        # The error messages are detailed to help debugging
+        def validate_abstract_values(
+            outvals: list, expected_outvals: list, branch_type: str, index: int = None
+        ) -> None:
+            """Ensure the collected abstract values match the expected ones."""
+
+            assert len(outvals) == len(expected_outvals), (
+                f"Mismatch in number of output variables in {branch_type} branch"
+                f"{'' if index is None else ' #' + str(index)}: "
+                f"{len(outvals)} vs {len(expected_outvals)}"
+            )
+            for i, (outval, expected_outval) in enumerate(zip(outvals, expected_outvals)):
+                assert outval == expected_outval, (
+                    f"Mismatch in output abstract values in {branch_type} branch"
+                    f"{'' if index is None else ' #' + str(index)} at position {i}: "
+                    f"{outval} vs {expected_outval}"
+                )
+
+        outvals_true = jaxpr_true.out_avals
+
+        if jaxpr_false is not None:
+            outvals_false = jaxpr_false.out_avals
+            validate_abstract_values(outvals_false, outvals_true, "false")
+
+        for idx, jaxpr_elif in enumerate(jaxpr_elifs):
+            outvals_elif = jaxpr_elif.out_avals
+            validate_abstract_values(outvals_elif, outvals_true, "elif", idx)
+
+        # We return the abstract values of the true branch since the abstract values
+        # of the false and elif branches (if they exist) should be the same
+        return outvals_true
+
+    return cond_prim
+
+
+def _capture_cond(condition, true_fn, false_fn=None, elifs=()) -> Callable:
+    """Capture compatible way to apply conditionals."""
+
+    import jax  # pylint: disable=import-outside-toplevel
+
+    cond_prim = _get_cond_qfunc_prim()
+
+    elifs = (elifs,) if len(elifs) > 0 and not isinstance(elifs[0], tuple) else elifs
+
+    @wraps(true_fn)
+    def new_wrapper(*args, **kwargs):
+
+        jaxpr_true = jax.make_jaxpr(functools.partial(true_fn, **kwargs))(*args)
+        jaxpr_false = (
+            (jax.make_jaxpr(functools.partial(false_fn, **kwargs))(*args) if false_fn else None)
+            if false_fn
+            else None

```

Comment:
There is a double `if false_fn else None` here.

```suggestion
            jax.make_jaxpr(functools.partial(false_fn, **kwargs))(*args)
            if false_fn
            else None
```

Reply:
Good catch!